### PR TITLE
[Gitpod CLI] Auto-install Docker when it's not available in the system

### PR DESCRIPTION
**Pull request by felladrin**
## Description

Auto-install Docker when it's not available in the system.

## Related Issue(s)

- https://github.com/gitpod-io/gitpod/pull/15740

## How to test

Open a workspace which uses an image (e.g. [ubuntu](https://hub.docker.com/_/ubuntu) / [busybox](https://hub.docker.com/_/busybox)) without Docker installed (e.g. https://github.com/gitpod-samples/template-elixir), and confirm if `gp rebuild` works.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by felladrin** [comment]
Hmm, it may not work for all cases. I've got this when running https://github.com/gitpod-samples/template-elixir on the Preview environment of this PR:

```
gitpod@gitpodsampl-templateeli-whzub6wvyma:/workspace/template-elixir$ curl -fsSL https://get.docker.com/rootless -o get-docker.sh
gitpod@gitpodsampl-templateeli-whzub6wvyma:/workspace/template-elixir$ sh get-docker.sh 
# Installing stable version 20.10.22
# Executing docker rootless install script, commit: 1d97565
Aborting because rootful Docker is running and accessible. Set FORCE_ROOTLESS_INSTALL=1 to ignore.
gitpod@gitpodsampl-templateeli-whzub6wvyma:/workspace/template-elixir$ FORCE_ROOTLESS_INSTALL=1 sh get-docker.sh 
# Installing stable version 20.10.22
# Executing docker rootless install script, commit: 1d97565
get-docker.sh: 159: lsmod: not found
grep: /lib/modules/5.15.0-47-generic/modules.builtin: No such file or directory
# Missing system requirements. Please run following commands to
# install the requirements and run this installer again.
# Alternatively iptables checks can be disabled with SKIP_IPTABLES=1

cat <<EOF | sudo sh -x
apt-get install -y uidmap
apt-get install -y iptables
modprobe ip_tables
EOF
```

To install some dependencies, it would require sudo. In that case, it would be worth having the option to run with the gitpod base image.

File: components/gitpod-cli/cmd/rebuild.go [hunk 1]
```diff
@@ -134,9 +136,42 @@ func runRebuild(ctx context.Context, supervisorClient *supervisor.SupervisorClie
...
+		dockerInstallCmd := exec.CommandContext(ctx, "sh", dockerInstallScript)
+		dockerInstallCmd.Env = append(dockerInstallCmd.Environ(), "FORCE_ROOTLESS_INSTALL=1")
+		dockerInstallCmd.Stdout = os.Stdout
+		dockerInstallCmd.Stderr = os.Stderr
+		err = dockerInstallCmd.Run()
+		if err != nil {
+			return utils.Outcome_SystemErr, err
+		}
+
+		err = os.Remove(dockerInstallScript)
```
Comment (akosyakov) [inline components/gitpod-cli/cmd/rebuild.go:171]: no need, the entire tmp dir will be removed.

**Comment by felladrin** [comment]
It now checks if the user is running as root or if the user has at least access to "sudo" before running the docker install.
We're not using the rootless-script anymore.
What do you think?

**Comment by felladrin** [comment]
ℹ️ We're considering closing this PR in favor of another approach: to restart the workspace using the base Gitpod image, which already has docker installed.

**Pull request closed without merging**